dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,247 @@
+﻿#include "pch.h"
+#include "NumberBoxParser.h"
+
+
+MathToken::MathToken(TokenType t, std::string s)
+{
+    this->type = t;
+    this->str = s;
+}
+
+MathToken::MathToken()
+{
+    this->type = MathToken::TokenType::EOFToken;
+    this->str = "";
+}
+
+MathTokenizer::MathTokenizer(std::string input)
```

Comment:
>string [](start = 34, length = 6)

this should be working with wstring -- Windows strings are all unicode (UTF+16). Generally it just means do the same as you were doing but put L at the front of chars and string literals (e.g. L'.' and L"...").

Reply:
Was planning on this though wasn't sure how many conversion hoops I'd have to jump through to do it so I left it out initially. Are most of the other properties for hstrings the same? Compatibility with stringstreams, Regex? I assume I'll just have to to_string for those.  @jevansaks

Reply:
Where did the string come from? hstring is a UTF-16 string, if you just assign it to a std::wstring_view then you'll incur no copy or conversions. Looks like there's std::wregex and friends for interacting with unicode strings.

Reply:
It comes from a TextBox so it is by default a wstring.  I'll look into the regex library and hopefully it should be a simple port.